lib/registry-sync/ecr-mirror.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  Construct, IAspect, IConstruct, Stack, Token,
+  aws_ecr as ecr,
+  aws_codebuild as codebuild,
+  aws_events as events,
+  aws_events_targets as targets,
+  aws_iam as iam,
+  aws_s3_assets as s3Assets,
+  aws_secretsmanager as sm,
+  custom_resources as cr,
+} from 'monocdk';
+import { MirrorSource } from './mirror-source';
+
+/**
+ * Authentication details for DockerHub.
+ *
+ * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#build-spec.env.secrets-manager
+ */
+export interface DockerHubCredentials {
+
+  /**
+   * The secret that contains the username and password for Dockerhub
+   */
+  readonly secret: sm.ISecret;
+
+  /**
+   * The secret key that contains the username in the specified secret.
+   */
+  readonly usernameKey: string;
+
+  /**
+   * The secret key that contains the password in the specified secret.
+   */
+  readonly passwordKey: string;
+
+  /**
+   * Version stage of the secret.
+   *
+   * @default 'AWSCURRENT'
+   */
+  readonly versionStage?: string;
+}
+
+/**
+ * Properties to initialize EcrRegistrySync
+ */
+export interface EcrMirrorProps {
+  /**
+   * The list of images to keep sync'ed.
+   */
+  readonly images: MirrorSource[];
+
+  /**
+   * Credentials to signing into Dockerhub.
+   */
+  readonly dockerHubCreds: DockerHubCredentials;
+
+  /**
+   * Sync job runs on a schedule.
+   * @default - does not run on schedule
+   */
+  readonly schedule?: events.Schedule;
+
+  /**
+   * Start the sync job immediately after the deployment.
+   * @default false
+   */
+  readonly autoStart?: boolean;
+}
+
+/**
+ * Synchronize images from DockerHub to an ECR registry in the AWS account.
+ * This is particularly useful to workaround DockerHub's throttling on pulls and use ECR instead.
+ */
+export class EcrMirror extends Construct {
+
+  private readonly _repos: Map<string, ecr.IRepository> = new Map();
```

Comment:
```suggestion
  private readonly _repos: Map<string, ecr.Repository> = new Map();
```

No need to potentially limit the API available to consumers, this will always be an owned repository.